### PR TITLE
Check if targets of links "out of root" exists

### LINKED ISSUE #58: Relative paths to application from user guide marked invalid
**Describe the bug**

We have some convenience links from our user guide to our full application. Our user guide is served along side our application, so we know what these relative URLs are when we build the application.

In `mkdocs-htmlproofer-plugin` `0.10.3` these were skipped just fine, but I suspect since:

- #49

These throw an error:

```
WARNING  -  Unable to locate source file for: ../../../api
ERROR    -  Error building page 'usage/rest.md':
ERROR    -  invalid url - ../../../api [404] [usage/rest.md]
```

**To Reproduce**

Add the following Markdown to the project:

```md
Browse the [`/api`](../../../api)
```

**Environment (please complete the following information):**
- OS (e.g. Windows 10/Ubuntu 20.04/macOS): macOS
- Python version (`python --version`): 3.10.8
- Package versions (`pip list`): 

```
argcomplete                2.1.2
beautifulsoup4             4.12.2
certifi                    2022.12.7
charset-normalizer         3.1.0
click                      8.1.3
colorama                   0.4.6
colorlog                   6.7.0
distlib                    0.3.6
filelock                   3.11.0
ghp-import                 2.1.0
idna                       3.4
Jinja2                     3.1.2
lxml                       4.9.2
Markdown                   3.3.7
MarkupSafe                 2.1.2
mergedeep                  1.3.4
mkdocs                     1.4.2
mkdocs-htmlproofer-plugin  0.12.0
mkdocs-macros-plugin       0.7.0
mkdocs-material            9.1.6
mkdocs-material-extensions 1.1.1
mkdocs-video               1.5.0
nox                        2022.11.21
nox-poetry                 1.0.2
packaging                  23.0
pip                        22.3.1
platformdirs               3.2.0
Pygments                   2.15.0
pymdown-extensions         9.11
python-dateutil            2.8.2
PyYAML                     6.0
pyyaml_env_tag             0.1
regex                      2023.3.23
requests                   2.28.2
setuptools                 65.6.3
six                        1.16.0
soupsieve                  2.4
termcolor                  2.2.0
tomlkit                    0.11.7
urllib3                    1.26.15
virtualenv                 20.21.0
watchdog                   3.0.0
wheel                      0.38.4
```

Contents of `mkdocs.yml`:

```yaml
...
plugins:
  - search
  - macros
  - offline:
      enabled: !ENV [MKDOCS_MATERIAL_OFFLINE, false]
  - mkdocs-video:
      is_video: True
      video_autoplay: True
      video_loop: True
      video_muted: True
      video_controls: False
      css_style:
        width: "100%"
  - htmlproofer:
      validate_external_urls: true
      raise_error: True
use_directory_urls: false
```

CC @kolibri91

### PR DESCRIPTION
**Pull request by kolibri91**
TBD

Closes #58

### CODE REVIEW

File: tests/integration/mkdocs.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-strict: True
+strict: False
```
Comment (johnthagen) [inline tests/integration/mkdocs.yml:1]: Will want to remove this before final merge.

Comment (kolibri91) [reply tests/integration/mkdocs.yml:1]: Of course, but I'm wondering you got it work within your example.
I normally get this warning:

```
WARNING  -  Documentation file 'index.md' contains a link to '..\..\..\README.md' which is not found in the documentation files.
```

**Comment by kolibri91** [comment]
> > [Handle python 3.6 with mkdocs < 1.4](https://github.com/manuzhang/mkdocs-htmlproofer-plugin/pull/59/commits/7a6f1fa072139bf2751d81612496e381c3f1d9ec)
> 
> In my opinion Python 3.6 is end of life and we should simply drop it from the test suite. Even Python 3.7 is almost end of life (2 months).
> 
>     * https://endoflife.date/python

It's not up to me to decide...

**Comment by johnthagen** [comment]
@manuzhang Would you have any issues with dropping Python 3.6 support since that is end of life?

**Comment by manuzhang** [comment]
> @manuzhang Would you have any issues with dropping Python 3.6 support since that is end of life?

No, but let's do it in a separate PR and update test workflows, README, etc.

File: tests/integration/mkdocs.yml (hunk 1 above)
Comment (kolibri91) [reply tests/integration/mkdocs.yml:1]: > Of course, but I'm wondering you how got it work within your example. I normally get this warning:
> 
> ```
> WARNING  -  Documentation file 'index.md' contains a link to '..\..\..\README.md' which is not found in the documentation files.
> ```

@johnthagen Ok, got it. `mkdocs` complains about missing files but not folders. So, your particular example from #58 works in `strict` mode.

**Pull request closed without merging**